templates/CHP.iesopt.template.yaml

```
@@ -0,0 +1,221 @@
+        # -------------------------------------------------
+        p_max = this.get("power_max")
+        h_max = this.get("heat_max")
+        cm = this.get("power_ratio")
+
+        if isnothing(p_max)
+            power_max = cm * h_max
+            this.set("power_max", power_max)
+        end
+        
+        if isnothing(h_max)
+            # Currently, nothing to do here (`h_max` is not actively used in backpressure mode).
+        end
+        
+        if isnothing(cm)
+            cm = p_max / h_max
+            this.set("power_ratio", cm)
+        end
+        # -------------------------------------------------
+
+        # -------------------------------------------------
+        # Prepare: eta_p, eta_h
+        # -------------------------------------------------
+        eta = this.get("efficiency_backpressure")
+        eta_h = eta / (1.0 + cm)
+        eta_p = cm * eta_h
+        # -------------------------------------------------
+
+        this.set("_turbine_outputs", "{electricity: $(to_p), heat: $(to_h)}")
+        this.set("_turbine_conversion", "1 $(fuel) -> $(eta_p) electricity + $(eta_h) heat")
+    elseif this.get("mode") == "condensing"
+        eta_p = this.get("efficiency_condensing")
+        this.set("_turbine_outputs", "{electricity: $(to_p)}")
+        this.set("_turbine_conversion", "1 $(fuel) -> $(eta_p) electricity")
+    elseif this.get("mode") == "extraction"
+        # -------------------------------------------------
+        # Prepare: power_max, heat_max, power_ratio, power_loss_ratio
+        # -------------------------------------------------
+        p_max = this.get("power_max")
+        h_max = this.get("heat_max")
+        p_at_h_max = this.get("power_at_heat_max")
+        cm = this.get("power_ratio")
+        cv = this.get("power_loss_ratio")
+        
+        if isnothing(p_max)
+            p_max = isnothing(p_at_h_max) ? (cm * h_max + cv * h_max) : (p_at_h_max + cv * h_max)
+            this.set("power_max", p_max)
+        end
+        
+        if isnothing(h_max)
+            h_max = isnothing(p_at_h_max) ? (p_max / (cm + cv)) : (p_at_h_max / cm)
+            this.set("heat_max", h_max)
+        end
+        
+        if isnothing(cm)
+            cm = p_at_h_max / h_max
+            this.set("power_ratio", cm)
+        end
+        
+        if isnothing(cv)
+            cv = (p_max - p_at_h_max) / h_max
+            this.set("power_loss_ratio", cv)
+        end
+        # -------------------------------------------------
+
+        # -------------------------------------------------
+        # Prepare: eta_p
+        # -------------------------------------------------
+        eta_p = nothing
+        eta_cd = this.get("efficiency_condensing")
+        eta_bp = this.get("efficiency_backpressure")
+        
+        if !isnothing(eta_cd) && isnothing(eta_bp)
+            eta_p = eta_cd
+        elseif isnothing(eta_cd) && !isnothing(eta_bp)
+            eta_p = eta_bp / (1.0 + cm)
```

Please check. I arrive at:
```suggestion
            eta_p = eta_bp * cm / (1.0 + cm)
```